docs/content/guide/component.ngdoc

```diff
@@ -0,0 +1,453 @@
+@ngdoc overview
+@name Components
+@sortOrder 305
+@description
+
+# Understanding Components
+
+In Angular, a Component is a special kind of {@link guide/directive directive} that uses a simpler
+configuration which is suitable for a component-based application structure.
+
+This makes it easier to write an app in a way thats similar to using Web Components or using Angular
+2's style of application architecture.
+
+Advantages of Components:
+- simpler configuration than plain directives
+- promote sane defaults and best practices
+- optimized for component-based architecture
+- writing component directives will make it easier to upgrade to Angular 2
+
+When not to use Components:
+
+- for directives that rely on DOM manipulation, adding event listeners etc, because the compile
+and link functions are unavailable
+- when you need advanced directive defintion options like priority, terminal, multi-element
+
```

Comment:
when you want a directive that is triggered by an attribute or CSS class, rather than an element